Client/Application/Shortcuts/ActivityShortcutManager.swift

```diff
@@ -0,0 +1,206 @@
+
+                switch BraveVPN.vpnState {
+                    case .notPurchased, .purchased, .expired:
+                        guard let enableVPNController = BraveVPN.vpnState.enableVPNDestinationVC else { return }
+                        
+                        bvc.openInsideSettingsNavigation(with: enableVPNController)
+                    case .installed(let connected):
+                        if !connected {
+                            BraveVPN.reconnect()
+                        }
+                }
+            case .openBraveNews:
+                if Preferences.BraveNews.isEnabled.value {
+                    bvc.openBlankNewTab(attemptLocationFieldFocus: false, isPrivate: false, isExternal: true)
+                    bvc.popToBVC()
+
+                    guard let newTabPageController = bvc.tabManager.selectedTab?.newTabPageViewController else { return }
+                    newTabPageController.scrollToBraveNews()
+                } else {
+                    let controller = BraveNewsSettingsViewController(dataSource: bvc.feedDataSource)
+                    let container = UINavigationController(rootViewController: controller)
+                    bvc.present(container, animated: true)
+                }
+            case .openPlayList:
+                let playlistController = (UIApplication.shared.delegate as? AppDelegate)?.playlistRestorationController ?? PlaylistViewController()
+                playlistController.modalPresentationStyle = .fullScreen
+                
+                bvc.present(playlistController, animated: true)
+        }
+    }
+    
+    // MARK: Intent Creation Methods
+    
+    private func createCustomIntent(for type: IntentType, with urlString: String) -> INIntent {
+        switch type {
+            case .openWebsite:
+                let intent = OpenWebsiteIntent()
+                intent.websiteURL = urlString
+                intent.suggestedInvocationPhrase = Strings.Shortcuts.customIntentOpenWebsiteSuggestedPhrase
+                
+                return intent
+            case .openHistory:
+                let intent = OpenHistoryWebsiteIntent()
+                intent.websiteURL = urlString
+                intent.suggestedInvocationPhrase = Strings.Shortcuts.customIntentOpenHistorySuggestedPhrase
+                
+                return intent
+            case .openBookmarks:
+                let intent = OpenBookmarkWebsiteIntent()
+                intent.websiteURL = urlString
+                intent.suggestedInvocationPhrase = Strings.Shortcuts.customIntentOpenBookmarkSuggestedPhrase
+                
+                return intent
+        }
+    }
+    
+    // MARK: Intent Donation Methods
+    
+    public func donateCustomIntent(for type: IntentType, with urlString: String) {
+        let intent = createCustomIntent(for: type, with: urlString)
+
+        let interaction = INInteraction(intent: intent, response: nil)
+        interaction.donate { error in
```

Comment:
nit: move the check for `isPrivateBrowsing` here instead of checking in callers?

Reply:
I want to leave the conditions inside the caller but in addition I added a private browsing guard here.